submissions/AnnaStoyano/oop-exercise/index.js

```diff
@@ -0,0 +1,101 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Mammal {
+   constructor(species, name, gender, legs, friends, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.friends = friends;
+      this.saying = saying;
+   }
+
+   tellAbout() {
+      const properties = ['species', 'name', 'gender']; //all mammals have this properties
+      for (let property in this) {
+         if (!properties.includes(property))
+            properties.push(property);
+      }
```

Comment:
A class method should work only with properties defined in the class.
### Why that?
A developer may introduce a child class having properties the developer may not want to be printed.
Current design will force the developer to change code in the base class. Other developers on the project may not be happy.
### Solution
1. Make the method to work with properties known to this class.
2. In every child class overload the method and make it call `super.tellAbout()` and supplement the result with the values from properties know to that child class.

Check [this doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/super) out.